directory/directory_dest.go

```diff
@@ -21,13 +21,14 @@ const version = "Directory Transport Version: 1.1\n"
 var ErrNotContainerImageDir = errors.New("not a containers image directory, don't want to overwrite important data")
 
 type dirImageDestination struct {
-	ref      dirReference
-	compress bool
+	ref        dirReference
+	compress   bool
+	decompress bool
```

Comment:
Non-blocking: Just store a single `desiredLayerCompression types.LayerCompression` here, and compute it in the constructor; that eliminates the `compress == true && decompress == true` concern (and works however we decide to define the `SystemContext` options).